core/amber/src/main/python/core/runnables/network_receiver.py

```diff
@@ -64,8 +64,10 @@ def data_handler(command: bytes, table: Table) -> int:
             """
             data_header = PythonDataHeader().parse(command)
             payload_type = data_header.payload_type
-            if payload_type == "data":
+            if payload_type == "Data":
                 payload = DataFrame(table)
+            elif payload_type == "State":
+                payload = MarkerFrame(State().from_dict(table.to_pandas().iloc[0].to_dict()))
```

Comment:
it is not very readable of what this line is doing. consider add comments or put it into a named function.

Reply:
Moved to a named function.